compress_multiphoton/compression.py

```diff
@@ -2,50 +2,84 @@
-    # max_acceptable_intensity = 100000
-
-    # Make sure field is at least 32 bytes (int16 overflows if summed to itself)
-    scan = scan.astype(np.float32, copy=False)
-
-    # Create pixel values at each position in field
-    eps = 1e-4  # needed for np.round to not be biased towards even numbers (0.5 -> 1, 1.5 -> 2, 2.5 -> 3, etc.)
-    pixels = np.round((scan[:, :, :-1] + scan[:, :, 1:]) / 2 + eps)
-
-    pixels = pixels.astype(np.int16 if np.max(abs(pixels)) < 2**15 else np.int32)
-
-    # Compute a good range of pixel values (common, not too bright values)
-    unique_pixels, counts = np.unique(pixels, return_counts=True)
-    min_intensity = min(unique_pixels[counts > min_count])
-    max_intensity = max(unique_pixels[counts > min_count])
-    max_acceptable_intensity = min(max_intensity, max_acceptable_intensity)
-    pixels_mask = np.logical_and(pixels >= min_intensity, pixels <= max_acceptable_intensity)
-
-    # Select pixels in good range
-    pixels = pixels[pixels_mask]
-    unique_pixels, counts = np.unique(pixels, return_counts=True)
-
-    # Compute noise variance
-    variances = ((scan[:, :, :-1] - scan[:, :, 1:]) ** 2 / 2)[pixels_mask]
-    pixels -= min_intensity
-    variance_sum = np.zeros(len(unique_pixels))  # sum of variances per pixel value
-    for i in range(0, len(pixels), int(1e8)):  # chunk it for memory efficiency
-        variance_sum += np.bincount(
-            pixels[i : i + int(1e8)], weights=variances[i : i + int(1e8)], minlength=np.ptp(unique_pixels) + 1
-        )[unique_pixels - min_intensity]
-    unique_variances = variance_sum / counts  # average variance per intensity
-
-    # Compute quantal size (by fitting a linear regressor to predict the variance from intensity)
-    X = unique_pixels.reshape(-1, 1)
-    y = unique_variances
-    model = TheilSenRegressor()  # robust regression
-    model.fit(X, y)
+def _longest_run(bool_array):
+    step = np.diff(np.int8(bool_array), prepend=0, append=0)
+    on = np.where(step == 1)[0]
+    off = np.where(step == -1)[0]
+    i = np.argmax(off - on)
+    return slice(on[i], off[i])
+
+
+def compute_quantal_size(movie: np.array) -> dict:
+    """Calculate quantal size for a movie.
+
+    Args:
+        movie (np.array):  A movie in the format (height, width, time).
+
+    Returns:
+        output (dict): A dictionary with the following keys:
+            - 'model': The fitted TheilSenRegressor model.
+            - 'min_intensity': Minimum intensity used.
+            - 'max_intensity': Maximum intensity used.
+            - 'intensity_levels': Intensity levels.
+            - 'variance': Variances at intensity levels.
+            - 'quantal_size': Estimated quantal size.
+            - 'zero_level': Estimated ADC gain.
+    """
+    movie = movie.astype(np.int32, copy=False)
+    intensity = (movie[:, :, :-1] + movie[:, :, 1:] + 1) // 2
+    difference = movie[:, :, :-1] - movie[:, :, 1:]
+
+    CTS_THRESHOLD = 1000
```

Comment:
avoid acronyms. What's CTS? Perpahs `MIN_COUNTS`

Reply:
Also, this value should probably depend on the length of the movie. For short movies, perhaps it should be 50 or 100.